src/Tools/dotnet-stack/Symbolicate.cs

```diff
@@ -0,0 +1,426 @@
+        {
+            if (Directory.Exists(tempDirectory))
+            {
+                Directory.Delete(tempDirectory, true);
+            }
+        }
+
+        private static void Symbolicator(IConsole console, List<string> xmlList, string inputPath, string outputPath)
+        {
+            if (xmlList.Count == 0)
+            {
+                RemoveTempDirectory();
+                throw new FileNotFoundException("Xml file not found\n");
+            }
+
+            GetLineFromStack(console, xmlList, inputPath, outputPath);
+
+            RemoveTempDirectory();
+        }
+
+        internal sealed class StackTraceInfo
+        {
+            public string Type;
+            public string Method;
+            public string Param;
+            public string Assembly;
+            public string Token;
+            public string Offset;
+            public string Document;
+            public string Filepath;
+            public string Filename;
+            public string StartLine;
+            public string EndLine;
+        }
+
+        private static void GetLineFromStack(IConsole console, List<string> xmlList, string inputPath, string outputPath)
+        {
+            try
+            {
+                StreamWriter fswi = null;
+                if (!inputPath.EndsWith(".symbolicated"))
+                {
+                    fswi = new StreamWriter(new FileStream(inputPath + ".symbolicated", FileMode.Create, FileAccess.Write));
+                }
+
+                string output = string.Empty;
+                StreamWriter fswo = null;
+                if (outputPath != null)
+                {
+                    fswo = new StreamWriter(new FileStream(outputPath, FileMode.Create, FileAccess.Write));
+                    output = $"\nOutput: {outputPath}\n";
+                }
+
+                using StreamReader fsri = new StreamReader(new FileStream(inputPath, FileMode.Open, FileAccess.Read));
+                while (!fsri.EndOfStream)
+                {
+                    string line = fsri.ReadLine();
+                    // The stacktrace must have "at ... in ... token ..."
+                    if (!line.Contains("at ") || !line.Contains(" in ") || !line.Contains("token"))
+                    {
+                        fswo?.WriteLine(line);
+                        console.Out.WriteLine($"{line}");
+                        continue;
+                    }
+                    string ret = GetRegex(console, line, xmlList, fswi);
+                    fswo?.WriteLine(ret);
+                    console.Out.WriteLine($"{ret}");
+                }
+                fswo?.Close();
+                fswi?.Close();
+                console.Out.WriteLine($"{output}");
+            }
+            catch (Exception e)
+            {
+                console.Error.WriteLine(e.Message);
+            }
+        }
+
+        private static string GetRegex(IConsole console, string line, List<string> xmlList, StreamWriter fsw)
```

Comment:
This function seems to have a few side-effects unrelated to getting the regex. I'd rename the function to better reflect that. Something like `TrySymbolicateLine` or something similar, since it looks like it _can_ fail.

In a similar vein, it looks like this function has the logic for writing to one file, while `GetLineFromStack` has the logic for writing to other outputs. Having the print logic spread out makes it difficult to reason about what's being written where and when. In this case, the function name doesn't imply that it does any writing.

Preferably, `GetLineFromStack` might look something like this:
```csharp
outputWriter = outputPath.IsNullOrEmpty() ?
	/* open file next to input */ :
	/* open at output path */;
foreach (line in inputFile)
	outputWriter.Write(TrySymbolicateLine(line));
```

so all the writing happens in one place.